src/Components/Authorization/AuthForm.tsx

```diff
@@ -0,0 +1,109 @@
+import React from "react"
+
+import ForgotPasswordForm from "./ForgotPasswordForm"
+import LoginForm from "./LoginForm"
+import RegisterForm from "./RegisterForm"
+import { FormComponentType, FormFields, Mode } from "./Types"
+
+type Validator = (val: string) => string
+
+interface Props extends React.HTMLProps<HTMLDivElement> {
+  mode: Mode
+  signupIntent?: string
+  destination?: string
+  formFields?: FormFields
+}
+
+interface State extends React.HTMLProps<HTMLDivElement> {
+  mode?: Mode
+  formFields: FormFields
+}
+
+class AuthForm extends React.Component<Props, State> {
+  static defaultProps: Partial<Props> = {
+    signupIntent: "sign_up",
+    destination: "/personalize", // ?
+    formFields: {
+      email: { value: "", error: "" },
+      name: { value: "", error: "" },
+      password: { value: "", error: "" },
+    },
+  }
+
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      mode: this.props.mode,
+      formFields: this.props.formFields,
+    }
+    this.handleUpdateInput = this.handleUpdateInput.bind(this)
+    this.handleChangeMode = this.handleChangeMode.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
+
+  validators = {
+    name: val => (val.length > 0 ? "" : "Please enter your name"),
+    email: val => (val.indexOf("@") > -1 ? "" : "Please enter your email"),
+    password: val =>
+      val.length > 6 ? "" : "Password must be 6 characters long",
+  }
```

Comment:
It is common that a single input field has multiple validation rules. We should probably make this a collection of validation objects so it can show different error messages depending on the user input.

Reply:
@yuki24 - Are we using any React form libs anywhere in Reaction? 

@anandaroop - What are ya'll using in Volt? I remember seeing it a while back but might be worthwhile introducing here rather than rolling own validation.

Reply:
@damassi No, everything is a plain React input/form. Do you think it is something we should look into now?

Reply:
Def -- IMO there's no reason to roll your own form stuff these days with such nice / simple react libraries around that reduce so much complexity. I _think_ this is what Roop et al used: https://github.com/jaredpalmer/formik

Reply:
I can recommend [Formik](https://github.com/jaredpalmer/formik) as a solution to form handling & validation in React.

It's got a bit of learning curve, but much less than anything involving Redux-ifying your forms.

Reply:
🙄 thanks!